### PR TITLE
Check --version under different user

### PR DESCRIPTION
**Pull request by partouf**

### CODE REVIEW

**Comment by partouf** [comment]
need to test with tools and libs to see if it didnt break anything

File: .github/workflows/install.yml [hunk 1]
```diff
@@ -17,4 +17,4 @@ jobs:
       - name: Set up environment
         run: make ce
       - name: Install ${{ github.event.inputs.compilers }}
-        run: sudo bin/ce_install install ${{ github.event.inputs.compilers }}
+        run: sudo bin/ce_install --check_user nobody install ${{ github.event.inputs.compilers }}
```
Comment (mattgodbolt) [inline .github/workflows/install.yml:20]:
```suggestion
        run: sudo bin/ce_install --check-user nobody install ${{ github.event.inputs.compilers }}
```

File: update_compilers/install_compilers.sh [hunk 2]
```diff
@@ -15,6 +15,7 @@ else
     echo "Skipping install of nightly compilers"
 fi
 
-ce_install compilers
+ce_install --check_user nobody compilers
```
Comment (mattgodbolt) [inline update_compilers/install_compilers.sh:18]:
```suggestion
ce_install --check-user nobody compilers
```

File: bin/lib/installable/installable.py [hunk 3]
```diff
@@ -161,6 +161,19 @@ def save_version(self, exe: str, res_call: str):
 
         nightlies.update_version(fullpath.as_posix(), str(modified), res_call.split("\n", 1)[0], res_call)
 
+    def check_output_under_different_user(self):
+        if self.install_context.run_checks_as_user != "":
```
Comment (mattgodbolt) [inline bin/lib/installable/installable.py:165]:
```suggestion
        if self.install_context.run_checks_as_user:
```

File: bin/lib/installable/installable.py [hunk 4]
```diff
@@ -161,6 +161,19 @@ def save_version(self, exe: str, res_call: str):
...
+            call = ["/usr/bin/sudo", "-u", self.install_context.run_checks_as_user] + self.check_call
+            res_call = self.install_context.check_output(
+                call, env=self.check_env, stderr_on_stdout=self.check_stderr_on_stdout
+            )
+            return None
```
Comment (mattgodbolt) [inline bin/lib/installable/installable.py:170]: Should this return the `res_call` ?

**Review by mattgodbolt — COMMENTED** [review]
Love the idea; thanks!

File: bin/lib/installable/installable.py (hunk 4 above)
Comment (partouf) [reply bin/lib/installable/installable.py:170]: yess was a testing thing, thanks for pointing it out

**Pull request merged**